google/datalab/utils/_http.py

```diff
@@ -52,7 +52,7 @@ def __init__(self, status, content):
         error = error['errors'][0]
       self.message += ': ' + error['message']
     except Exception:
-      lines = content.split('\n') if isinstance(content, basestring) else []
+      lines = str(content).split('\n') if isinstance(content, basestring) else []
```

Comment:
This doesn't seem like the right way to handle unicode string, I think it must be decoded, otherwise the string isn't really split:
```
>>> print(str(b'hello\nworld').split('\n')) .            # ["b'hello\\nworld'"]
>>> print(b'hello\nworld'.decode('utf-8').split('\n'))   # ['hello', 'world']
```

Reply:
sorry about the slow response.. yes you're right of course.. wasn't sure that I can assume the utf8 encoding so a string cast is the most "robust" implementation (i.e. not crashing the code when encountering bytes outside the charset but can end up with garbled messages). If we can assume utf8 yes you're right. I've updated the code and pushed the change